spec/controllers/users/confirmations_controller_spec.rb

```diff
@@ -95,6 +95,20 @@
 
       patch :confirm, password_form: { password: 'NewVal' }, confirmation_token: 'foo'
     end
+
+    context 'user supplies invalid password' do
+      render_views
+
+      it 'includes invalid password feedback message' do
+        user = create(:user, :unconfirmed)
+        user.update(confirmation_token: 'foo')
+
+        patch :confirm, password_form: { password: 'password' }, confirmation_token: 'foo'
+
+        expect(response).to_not redirect_to phone_setup_url
+        expect(response.body).to match('not strong enough')
```

Comment:
Since this is controller, the feedback is good but also test for status code and user values not being updated.